gnuradio-runtime/lib/constants.cc.in

```diff
@@ -29,8 +29,12 @@ const std::string prefix()
     std::filesystem::path prefix_rel_lib = "@prefix_relative_to_lib@";
     std::filesystem::path gr_runtime_lib_path = boost::dll::this_line_location().string();
     // Normalize before decomposing path so result is reliable
-    std::filesystem::path prefix_path =
-        gr_runtime_lib_path.lexically_normal().parent_path() / prefix_rel_lib;
+    std::filesystem::path partial_path = gr_runtime_lib_path.lexically_normal().parent_path();
+    // If this is a symlink, dereference (/bin ==> /usr/bin on some systems)
+    if (std::filesystem::is_symlink(partial_path)) {
+        partial_path = partial_path.parent_path() / std::filesystem::read_symlink(partial_path);
+    }
+    std::filesystem::path prefix_path = partial_path / prefix_rel_lib;
```

Comment:
I haven't tested this, but I think constructing a canonical path will be a more reliable way for resolving the symlink, because then we don't have to worry about whether the link is relative or absolute and handle the path splitting like you do.
```suggestion
    std::filesystem::path canonical_lib_path = std::filesystem::canonical(gr_runtime_lib_path);
    std::filesystem::path prefix_path = canonical_lib_path.parent_path() / prefix_rel_lib;
```

Reply:
Ah, `canonical` is C++17, so only a valid solution for the master branch?

Reply:
> Ah, `canonical` is C++17, so only a valid solution for the master branch?

Nevermind, all of `std::filesystem` is C++17, and we can just use the boost equivalent for `maint-3.9`. So I'm back to recommending `canonical` as a more robust solution.

Reply:
I've just tested your suggested version and it's not only shorter and cleaner, but it also works.  Let's commit your version.